src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -271,14 +271,35 @@ private void launchNode(SchedulerDriver driver, Offer offer,
           .setSlaveId(offer.getSlaveId())
           .addAllResources(taskResources)
           .setData(ByteString.copyFromUtf8(
-              String.format("bin/hdfs-mesos-%s", taskName)))
+              String.format("bin/hdfs-mesos-%s", taskType)))
           .build();
       tasks.add(task);
 
-      liveState.addStagingTask(task.getTaskId());
-      persistentState.addHdfsNode(taskId, offer.getHostname(), taskName);
+      liveState.addStagingTask(task.getTaskId(), taskName);
+      persistentState.addHdfsNode(taskId, offer.getHostname(), taskType);
     }
     driver.launchTasks(Arrays.asList(offer.getId()), tasks);
+    return true;
+  }
+
+  private String getNodeName(String taskType) {
+    if (taskType.equals(HDFSConstants.NAME_NODE_ID)) {
+      for (int i = HDFSConstants.TOTAL_NAME_NODES; i > 0; i--) {
```

Comment:
Usability/confusability issue? Doesn't this mean that we'll launch a task named NN2 before launching NN1?
Seems like it would be easy enough to say `for (int i = 1; i <= TOTAL_NNs; i++)` to get 1-based naming.